### PR TITLE
chore: move *.cmake in cmake directory

### PR DESCRIPTION
**Pull request by dmehala**
Changes:
- Update nginx-module.cmake to add `--with-threads` only when ASM is enabled.
- Default back to NGINX 1.26.0 the dev workflow.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -145,18 +145,18 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   set_target_properties(ngx_http_datadog_module PROPERTIES LINK_DEPENDS ${LINKER_SCRIPT})
   target_link_options(ngx_http_datadog_module PRIVATE "-Wl,--version-script=${LINKER_SCRIPT}")
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-  set(EXPORTED_SYMBOLS ${CMAKE_SOURCE_DIR}/nginx_module.exported)
+  set(EXPORTED_SYMBOLS ${CMAKE_SOURCE_DIR}/nginx_module.ld)
```
Comment (dmehala) [inline CMakeLists.txt:148]: I reused the `.ld` but it's up to you @cataphract

File: README.md [hunk 2]
```diff
@@ -51,14 +51,19 @@ Supported architectures (`<arch>`) are `amd64` and `arm64`.
...
+### Tracing
 - Connect to the Datadog agent at `http://localhost:8126`.
 - Create one span per request:
     - Service name is "nginx".
     - Operation name is "nginx.request".
     - Resource name is `"$request_method $uri"`, e.g. "GET /api/book/0-345-24223-8/title".
     - Includes multiple `http.*` [tags][8].
 
+### Appsec
+TBD
```
Comment (dmehala) [inline README.md:65]: - [ ] TODO

File: CMakeLists.txt (hunk 1 above)
Comment (cataphract) [reply CMakeLists.txt:148]: The reason there are separate ones is because iirc the mac os linker doesn't support the version script format in the .ld file

Comment (pablomartinezbernardo) [reply CMakeLists.txt:148]: I was trying to build things on mac from `master` and was getting this error

`gmake[3]: *** No rule to make target '/Users/pablo.martinez/dev/nginx-datadog/nginx_module.exported', needed by 'ngx_http_datadog_module.so'.  Stop.`

Applying this suggestion makes my build succeed.

**Review by pablomartinezbernardo — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (dmehala) [inline README.md:65]:
```suggestion
```

**Pull request merged**